app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,235 @@
+    {
+        $user_role_id = 3;
+        $role_id = $request->input("role_id") != null ?
+            $request->input("role_id") : $user_role_id;
+        if (!$this->checkRole($role_id)) {
+            return response('Bad request', 400);
+        }
+        $user = new User($request->all());
+        $user->role_id = $role_id;
+        return $user->save();
+    }
+
+    /**
+     * @OA\Get(
+     *      path="/users/{id}",
+     *      operationId="getUserById",
+     *      tags={"Users"},
+     *      summary="Get user information",
+     *      description="Returns user data",
+     *      @OA\Parameter(
+     *          name="id",
+     *          description="User id",
+     *          required=true,
+     *          in="path",
+     *          @OA\Schema(
+     *              type="integer"
+     *          )
+     *      ),
+     *      @OA\Response(
+     *          response=200,
+     *          description="Successful operation",
+     *          @OA\JsonContent(ref="#/components/schemas/UserResource")
+     *       ),
+     *      @OA\Response(
+     *          response=400,
+     *          description="Bad Request"
+     *      ),
+     *      @OA\Response(
+     *          response=401,
+     *          description="Unauthenticated",
+     *      ),
+     *      @OA\Response(
+     *          response=403,
+     *          description="Forbidden"
+     *      )
+     * )
+     */
+    public function show($id)
+    {
+        return new UserResource(User::findOrFail($id));
+    }
+
+    /**
+     * @OA\Put(
+     *      path="/users/{id}",
+     *      operationId="updateUser",
+     *      tags={"Users"},
+     *      summary="Update existing user",
+     *      description="Returns updated user data",
+     *      @OA\Parameter(
+     *          name="id",
+     *          description="User id",
+     *          required=true,
+     *          in="path",
+     *          @OA\Schema(
+     *              type="integer"
+     *          )
+     *      ),
+     *      @OA\RequestBody(
+     *          required=true,
+     *          @OA\JsonContent(ref="#/components/schemas/UpdateUserResource")
+     *      ),
+     *      @OA\Response(
+     *          response=202,
+     *          description="Successful operation",
+     *          @OA\JsonContent(ref="#/components/schemas/UserResource")
+     *       ),
+     *      @OA\Response(
+     *          response=400,
+     *          description="Bad Request"
+     *      ),
+     *      @OA\Response(
+     *          response=401,
+     *          description="Unauthenticated",
+     *      ),
+     *      @OA\Response(
+     *          response=403,
+     *          description="Forbidden"
+     *      ),
+     *      @OA\Response(
+     *          response=404,
+     *          description="Resource Not Found"
+     *      )
+     * )
+     */
+    public function update(Request $request, int $id)
+    {
+        $user = User::findOrFail($id);
```

Comment:
do poprawy

Reply:
Chodzi, żeby zmienić to int $id na usera?

Reply:
https://youtu.be/ShrS3HXwzPg?t=63

Reply:
https://www.youtube.com/watch?v=6dEfxGLgevM&t=244s tutaj jak to działa

Reply:
Git :)